app/lib/clients/vault/oidc.rb

```diff
@@ -0,0 +1,151 @@
+=begin
+
+The purpose of this module is to assign a policy to an OIDC user, by
+mapping that user's email address to a policy we create.
+It works as follows:
+
+It creates an OIDC provider and user.  That user has a
+username/password/email addr, that can be accessed with OIDC auth.
+
+It creates an OIDC client which connects to that provider.  When a
+user tries to auth, the client connects to the provider, which opens
+up a browser window allowing the user to enter his username/password.
+
+On success, the provider returns an OIDC token, which includes the
+user's email addr.
+
+The client has been configured to map that email address to an entity
+in vault, which has the policy which we want the user to have.
+
+So the mapping goes from the email address on the provider, to the
+policy in vault.  The email addr may not be the best mapping to use.
+Some other piece of user info may ultimately be better used to map the
+user to the policy.  But we don't yet have a good understanding of
+different OIDC provider configurations, so this should be good enough
+for now
+
+Note that this provider is only meant to be used in our dev/test
+environment to excercise the client.  In a prod env, a real OIDC
+provider is configured in.
```

Comment:
did you mean to reference a file here?

Reply:
What I meant was that they would fill in the config here in the astral.yml file: https://github.com/G-Research/astral/blob/3d3d1fb26f0eb9a3042b4c27b2df0a4f4dcb439b/config/astral.yml#L43-L46

Are you saying I should make that explicit?

Reply:
I added the astral.yml file name